examples/platform_channel/ios/Runner/AppDelegate.m

```diff
@@ -13,12 +13,11 @@ @implementation AppDelegate {
 - (BOOL)application:(UIApplication*)application
     didFinishLaunchingWithOptions:(NSDictionary*)launchOptions {
   [GeneratedPluginRegistrant registerWithRegistry:self];
-  FlutterViewController* controller =
-      (FlutterViewController*)self.window.rootViewController;
+  NSObject<FlutterPluginRegistrar>* registrar = [self registrarForPlugin:@"battery"];
```

Comment:
"plugin"? maybe a new method like `[self createMessenger:@"batter"]`?

Reply:
Hold up, this is still a draft.  I'm not proposing anything yet =)

Reply:
Another option (I'm not proposing) is that they could subclass FlutterViewController and initialize the platform channels in `awakeFromNib`.

Reply:
> "plugin"? maybe a new method like `[self createMessenger:@"battery"]`?

I've updated the design doc to hopefully address this question.